orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/expressions/ExpressionTransform.java

```diff
@@ -270,21 +270,31 @@ static String escapeSimpleExpression(String expression) {
    * @param e #stage('property') becomes #stage(#root.execution, 'property')
    * @return an execution aware helper function
    */
-  private String includeExecutionParameter(String e) {
+  // Visible for testing
+  String includeExecutionParameter(String e) {
     String expression = e;
 
     // An expression aware function is any that takes an Execution as its first parameter
     Set<String> expressionAwareFunctions = new HashSet<>(EXECUTION_AWARE_FUNCTIONS);
-    expressionFunctionProviders.forEach(p -> {
-      p.getFunctions().forEach(f -> {
+    for (ExpressionFunctionProvider p : expressionFunctionProviders) {
+      String namespace = p.getNamespace();
+      for (ExpressionFunctionProvider.FunctionDefinition f : p.getFunctions()) {
         if (!f.getParameters().isEmpty() && f.getParameters().get(0).getType() == Execution.class) {
-          expressionAwareFunctions.add(f.getName());
+          String functionName = f.getName();
+          if (namespace != null) {
+            String registeredFunction = namespace + "_" + functionName;
+            String nameSpaceFunction = namespace + "." + functionName;
+            expression = expression.replaceAll("#" + nameSpaceFunction + "\\(", "#" + registeredFunction + "(");
```

Comment:
We feel that the consistency of using ‘.’ instead of ‘_’ as a delimiter in the incoming SpEL expression is consistent with the use of ‘.’ as a delimiter in Javascript (for example), and will thus lead to a better user experience.  Having the pattern of using ‘.’ in Deck (and in pipeline JSON), then transforming the ‘.’ into ‘_’ under the covers in Orca feels like a small technical and conceptual price to pay for a syntax which will pay dividends in the future.

Reply:
Seems reasonable to me. Can we get rid of `_`? So there is a true One Way to do things?

Reply:
Unfortunately, I do not know of a way to tell the SpEL parser that the namespacing dots should not be considered a delimiter which introduces another token.

Reply:
My preference is to only support one way of using a namespace'd fn and it sounds like that needs to be `_` delimitated (rather than have to worry about converting back and forth between implementation and documentation).

Will merge after.

Reply:
There is already some precedent for using . to delimit some scope at https://github.com/spinnaker/deck/blob/0333b67defc62ae77cf5e48096ce7ea4715c1a5b/app/scripts/modules/core/src/widgets/spelText/SpelAutocompleteService.ts#L54..L55.  I think that changing those names without understanding its impact is beyond the scope of our namespacing change.  Changing our implementation to support _ would make this inconsistent.

Reply:
We have decided to remove namespacing from our change, in lieu of @marchello2000 's initial suggestion of `cfServiceKey` instead of either `cloudfoundry_serviceKey` or `cloudfoundry.serviceKey`.  If `_` is going to be used for namespacing, then nothing would really prevent us from using `_` as part of the function name in Orca.  This invites the question of why we would use namespaces, since there are now two ways to define a function which is exposed in Deck with `_` contained within its name.